cachi2/core/package_managers/generic/main.py

```diff
@@ -0,0 +1,81 @@
+import logging
+
+import yaml
+from pydantic import ValidationError
+
+from cachi2.core.errors import PackageManagerError, PackageRejected
+from cachi2.core.models.input import Request
+from cachi2.core.models.output import RequestOutput
+from cachi2.core.models.sbom import Component
+from cachi2.core.package_managers.generic.models import GenericLockfileV1
+from cachi2.core.rooted_path import RootedPath
+
+log = logging.getLogger(__name__)
+DEFAULT_LOCKFILE_NAME = "generic_lockfile.yaml"
+DEFAULT_DEPS_DIR = "deps/generic"
+
+
+def fetch_generic_source(request: Request) -> RequestOutput:
+    """
+    Resolve and fetch generic dependencies for a given request.
+
+    :param request: the request to process
+    """
+    components = []
+    for package in request.generic_packages:
+        path = request.source_dir.join_within_root(package.path)
+        components.extend(_resolve_generic_lockfile(path, request.output_dir))
+    return RequestOutput.from_obj_list(components=components)
+
+
+def _resolve_generic_lockfile(source_dir: RootedPath, output_dir: RootedPath) -> list[Component]:
+    """
+    Resolve the generic lockfile and pre-fetch the dependencies.
+
+    :param source_dir: the source directory to resolve the lockfile from
+    :param output_dir: the output directory to store the dependencies
+    """
+    lockfile_path = source_dir.join_within_root(DEFAULT_LOCKFILE_NAME)
+    if not lockfile_path.path.exists():
+        raise PackageRejected(
+            f"Cachi2 generic lockfile '{DEFAULT_LOCKFILE_NAME}' missing, refusing to continue.",
+            solution=(
+                f"Make sure your repository has cachi2 generic lockfile '{DEFAULT_LOCKFILE_NAME}' checked in "
+                "to the repository."
+            ),
+        )
+
+    log.info(f"Reading generic lockfile: {lockfile_path}")
+    lockfile = _load_lockfile(lockfile_path)
+    for artifact in lockfile.artifacts:
+        log.info(f"Resolving artifact: {artifact.download_url}")
```

Comment:
This one should be a debug-level log rather than info.

Reply:
Fixed.